DigitalLearningSolutions.Web.Tests/Controllers/Login/LoginControllerTests.cs

```diff
@@ -0,0 +1,153 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.Login
+{
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.Helpers;
+    using DigitalLearningSolutions.Web.Controllers;
+    using DigitalLearningSolutions.Web.Tests.TestHelpers;
+    using DigitalLearningSolutions.Web.ViewModels.Login;
+    using FakeItEasy;
+    using FluentAssertions.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    internal class LoginControllerTests
+    {
+        private LoginController controller;
+        private ICryptoService cryptoService;
+        private IUserService userService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            userService = A.Fake<IUserService>();
+            cryptoService = A.Fake<ICryptoService>();
+
+            A.CallTo(() => userService.GetAdminUserByUsername(A<string>._))
+                .Returns(UserTestHelper.GetDefaultAdminUser());
+            A.CallTo(() => userService.GetDelegateUsersByUsername(A<string>._))
+                .Returns(new List<DelegateUser> { UserTestHelper.GetDefaultDelegateUser() });
+            A.CallTo(() => cryptoService.VerifyHashedPassword(A<string>._, A<string>._))
+                .Returns(true);
+
+            controller = LoginTestHelper.GetLoginControllerWithUnauthenticatedUser(userService, cryptoService);
+        }
+
+        [Test]
+        public void Index_should_render_if_user_is_unauthenticated()
+        {
+            // When
+            var result = controller.Index();
+
+            // Then
+            result.Should().BeViewResult().WithDefaultViewName();
+        }
+
+        [Test]
+        public void Index_should_redirect_if_user_is_authenticated()
+        {
+            // Given
+            var controllerWithAuthenticatedUser =
+                LoginTestHelper.GetLoginControllerWithAuthenticatedUser(userService, cryptoService);
+
+            // When
+            var result = controllerWithAuthenticatedUser.Index();
+
+            // Then
+            result.Should().BeRedirectToActionResult()
+                .WithControllerName("Home").WithActionName("Index");
+        }
+
+        [Test]
+        public void Successful_sign_in_should_render_home_page()
```

Comment:
Could we set up the test such that the user service & crypto service mocks return what's expected (e.g. approved user and true)

Reply:
In the SetUp, I set up the mocks to return what's expected for most of the tests, and then when those tests differ in requirements I change the mock return in the tests themselves. Would we prefer them being defined in every test then?